fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -77,58 +86,56 @@ func startCommand(cmd *exec.Cmd) error {
 	return nil
 }
 
-func getNewClientID(admin servicesPb.AdminClient, startTime time.Time) (string, error) {
+func getNewClientIDs(admin servicesPb.AdminClient, startTime time.Time) ([]string, error) {
 	var ids [][]byte
 	ctx := context.Background()
 	res, err := admin.ListClients(ctx,
 		&servicesPb.ListClientsRequest{ClientIds: ids},
 		grpc.MaxCallRecvMsgSize(1024*1024*1024))
 	if err != nil {
-		return "", fmt.Errorf("ListClients RPC failed: %v", err)
-	}
-	if len(res.Clients) == 0 {
-		return "", errors.New("No new clients")
+		return nil, fmt.Errorf("ListClients RPC failed: %v", err)
 	}
-	var client *servicesPb.Client
-	lastVisitTime := startTime
+
+	var newClients []string
 	for _, cl := range res.Clients {
-		curLastVisitTime, err := ptypes.Timestamp(cl.LastContactTime)
+		lastContactTime, err := ptypes.Timestamp(cl.LastContactTime)
 		if err != nil {
 			continue
 		}
-		if curLastVisitTime.After(lastVisitTime) {
-			client = cl
-			lastVisitTime = curLastVisitTime
+		if lastContactTime.After(startTime) {
+			id, err := common.BytesToClientID(cl.ClientId)
```

Comment:
Why? There is `if err == nil {` on the next line. If there is an error we assume that it's not a newly connected client and just skip it. Otherwise, we append it to the newClients slice.

Reply:
Yes, I think that's not the case - BytesToClientID shouldn't fail. Added `if err != nil {` in the new commit. PTAL